---
title: "Gene expression and RNA Pol2 pausing of gene sets at wild type"
author: "Hani Kim"
date: "`r paste0('Initiated on 2020 Oct 31, compiled on ', format(Sys.time(), '%Y %b %d'))`"
output:
  html_document:
    code_folding: hide
    fig_height: 12
    fig_width: 12
    toc: yes
    number_sections: true
    toc_depth: 3
    toc_float: yes
    theme: paper
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Summary
This script generates boxplot visualisation for the expression and RNAP2 pausing signal at LSD1-specific, c-Myc specific and colocalised sites.

# Loading 
## Load libraries
```{r}
library(readxl)
```

## Load pausing index
```{r}
PI <- read.table("../../Data/PI_ES_0h.txt", header=F)
PI <- sapply(split(PI[,3], PI[,2]), mean)
names(PI) <- toupper(names(PI))
PIscore <- PI
```

## Target genes
```{r}
load("../../Data/TopGeneSets.Rdata")
```

# Process ESC transcriptome data
```{r}
rTranscriptome <- read_excel("../../Data/Transcriptome.xlsx", sheet = "Raw Transcriptome")
rownames <- rTranscriptome$`Ensembl~GN`
rownames <- strsplit(rownames, "~", 2)
rownames <- do.call(rbind, rownames)
rownames <- toupper(rownames[,2])
rTranscriptome <- as.matrix(rTranscriptome[-1])
rownames(rTranscriptome) <- rownames

genetofilter <- names(which(table(unlist(apply(log2(rTranscriptome)[,1:2], 2, function(x) {
    names(x[x < 5])
}))) == 2))

rTranscriptome.ave <- data.frame(
  T0 = rowMeans(rTranscriptome[,c(1,9)]),
  T1 = rowMeans(rTranscriptome[,c(2,10)]),
  T6 = rowMeans(rTranscriptome[,c(3,11)]),
  T12 = rowMeans(rTranscriptome[,c(4,12)]),
  T24= rowMeans(rTranscriptome[,c(5,13)]),
  T36 = rowMeans(rTranscriptome[,c(6,14)]),
  T48 = rowMeans(rTranscriptome[,c(7,15)]),
  T72 = rowMeans(rTranscriptome[,c(8,16)]))
  
colnames(rTranscriptome.ave) <- c("0hr", "1hr", "6hr", "12hr", "24hr", "36hr", "48hr", "72hr")
rTranscriptome.ave <- as.matrix(rTranscriptome.ave)
rownames(rTranscriptome.ave) <- rownames(rTranscriptome)

Transcriptome_0hr <- log2(rTranscriptome.ave[,1])
```

# Create list of geneSets
```{r}
t=list(myc=combined_cMycSpecific_rank_sorted,
       common=combined_common_rank_sorted,
       lsd1=combined_Lsd1Specific_rank_sorted)
```

# Plot pausing index
```{r}
o1 <- intersect(names(PIscore), t$myc)
o2 <- intersect(names(PIscore), t$common)
o3 <- intersect(names(PIscore), t$lsd1)

genetokeep <- names(PIscore) %in% rownames(rTranscriptome)[!rownames(rTranscriptome) %in% genetofilter]
par(mfrow=c(1,1), mar=c(2,2,2,2))
boxplot(PIscore[genetokeep],
        PIscore[o1],
        PIscore[o2],
        PIscore[o3], notch=TRUE, outline=F,
        names=c("all","myc", "common", "lsd1"), main="PIscore")
```

```{r}
wilcox.test(PIscore[o1], PIscore[o2]) #cmyc vs common
wilcox.test(PIscore[o1], PIscore[o3]) #cmyc vs lsd1
wilcox.test(PIscore[o2], PIscore[o3]) #common vs lsd1
```

# Plot ESC expression
```{r}
o1 <- intersect(names(Transcriptome_0hr), t$myc)
o2 <- intersect(names(Transcriptome_0hr), t$common)
o3 <- intersect(names(Transcriptome_0hr), t$lsd1)

genetokeep <- names(Transcriptome_0hr) %in% rownames(rTranscriptome)[!rownames(rTranscriptome) %in% genetofilter]
genetokeep1 <- names(Transcriptome_0hr) %in% rownames(rTranscriptome)[!rownames(rTranscriptome) %in% genetofilter] & 
  names(Transcriptome_0hr) %in% o1
genetokeep2 <- names(Transcriptome_0hr) %in% rownames(rTranscriptome)[!rownames(rTranscriptome) %in% genetofilter] & 
  names(Transcriptome_0hr) %in% o2
genetokeep3 <- names(Transcriptome_0hr) %in% rownames(rTranscriptome)[!rownames(rTranscriptome) %in% genetofilter] & 
  names(Transcriptome_0hr) %in% o3
```

```{r}
par(mfrow=c(1,1))
boxplot(Transcriptome_0hr[genetokeep],
        Transcriptome_0hr[o1],
        Transcriptome_0hr[o2],
        Transcriptome_0hr[o3], notch=TRUE, outline=F,
        names=c("all","myc", "common", "lsd1"), main="'all genes' filtered")
abline(h=median(Transcriptome_0hr[genetokeep]), col="red", lty="dotted", lwd=2)
```

```{R}
wilcox.test(Transcriptome_0hr[o1], Transcriptome_0hr[o2]) #cmyc vs common
wilcox.test(Transcriptome_0hr[o1], Transcriptome_0hr[o3]) #cmyc vs lsd1
wilcox.test(Transcriptome_0hr[o2], Transcriptome_0hr[o3]) #common vs lsd1
```

```{r}
par(mfrow=c(1,1))
boxplot(Transcriptome_0hr[genetokeep],
        Transcriptome_0hr[genetokeep1],
        Transcriptome_0hr[genetokeep2],
        Transcriptome_0hr[genetokeep3], notch=TRUE, outline=F,
        names=c("all","myc", "common", "lsd1"), main="all categories filtered")
abline(h=median(Transcriptome_0hr[genetokeep]), col="red", lty="dotted", lwd=2)

wilcox.test(Transcriptome_0hr[o1], Transcriptome_0hr[o2]) #cmyc vs common
wilcox.test(Transcriptome_0hr[o1], Transcriptome_0hr[o3]) #cmyc vs lsd1
wilcox.test(Transcriptome_0hr[o2], Transcriptome_0hr[o3]) #common vs lsd1
```

# Session Info
```{r}
sessioninfo::session_info()
```
