---
title: "Change in gene expression with LSD1 knockdown"
author: "Hani Kim"
date: "`r paste0('Initiated on 2020 Oct 31, compiled on ', format(Sys.time(), '%Y %b %d'))`"
output:
  html_document:
    code_folding: hide
    fig_height: 12
    fig_width: 12
    toc: yes
    number_sections: true
    toc_depth: 3
    toc_float: yes
    theme: paper
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Summary
This script generates boxplot visualisation for the change in RNAP2 pausing signal with LSD1 depletion.

# Loading 

## Target genes
```{r}
load("../../Data/targets.RData")
```

## Control and LSD1-depleted RNAP2 ChIP-seq data
```{r}
PI_NT_ESC <- read.delim2("../../Data/PI_NT_ESC.txt", header = TRUE)
PI_NT_ESC.numeric <- sapply(split(PI_NT_ESC[,3], toupper(PI_NT_ESC[,1])), function(x)as.numeric(x[1]))

PI_shLSD1_ESC <- read.delim2("../../Data/PI_shLSD1_ESC.txt", header = TRUE)
PI_shLSD1_ESC.numeric <- sapply(split(PI_shLSD1_ESC[,3], toupper(PI_shLSD1_ESC[,1])), function(x)as.numeric(x[1]))
```

# Boxplot visualisation
```{r}
o1 <- intersect(cMyc.targets, names(PI_NT_ESC.numeric))
o2 <- intersect(common.targets, names(PI_NT_ESC.numeric))
o3 <- intersect(Lsd1.targets, names(PI_NT_ESC.numeric))

boxplot(PI_NT_ESC.numeric[o1], 
        PI_NT_ESC.numeric[o2], 
        PI_NT_ESC.numeric[o3], 
        PI_NT_ESC.numeric, outline=F, notch=TRUE)
```

# ratio
```{R}
o <- intersect(names(PI_NT_ESC.numeric), names(PI_shLSD1_ESC.numeric))
PI_shLSD1_NT.ratio <- PI_shLSD1_ESC.numeric[o] / PI_NT_ESC.numeric[o]
PI_shLSD1_NT.ratio.normalised <- PI_shLSD1_NT.ratio - median(PI_shLSD1_NT.ratio) + 1

o1 <- intersect(cMyc.targets, names(PI_shLSD1_NT.ratio.normalised))
o2 <- intersect(common.targets, names(PI_shLSD1_NT.ratio.normalised))
o3 <- intersect(Lsd1.targets, names(PI_shLSD1_NT.ratio.normalised))

boxplot(PI_shLSD1_NT.ratio.normalised[o1], 
        PI_shLSD1_NT.ratio.normalised[o2], 
        PI_shLSD1_NT.ratio.normalised[o3], 
        PI_shLSD1_NT.ratio.normalised, 
        outline=F, notch=TRUE, col = RColorBrewer::brewer.pal(4, "Accent"),
        ylab="Change of Pausing (LSD1 KD vs NT)",
        names=c("cMyc-specific", "co-localize", "LSD1-specific", "All genes"), las=2)
```

# Session Info
```{r}
sessioninfo::session_info()
```
