---
title: "LSD1 signal by gene expression and pausing groups"
author: "Hani Kim"
date: "`r paste0('Initiated on 2020 Oct 31, compiled on ', format(Sys.time(), '%Y %b %d'))`"
output:
  html_document:
    code_folding: hide
    fig_height: 6
    fig_width: 5
    toc: yes
    number_sections: true
    toc_depth: 3
    toc_float: yes
    theme: paper
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Loading
## Load libraries
```{r}
suppressPackageStartupMessages({
    library(readxl)
    library(dplyr)
    library(ggplot2)
    library(readr)
})
```

## Load ESC expression data
```{r}
rTranscriptome <- read_excel("../../Data/Transcriptome.xlsx", sheet = "Raw Transcriptome")
rownames <- rTranscriptome$`Ensembl~GN`
rownames <- strsplit(rownames, "~", 2)
rownames <- do.call(rbind, rownames)
rownames <- toupper(rownames[,2])
rTranscriptome <- as.matrix(rTranscriptome[-1])
rownames(rTranscriptome) <- rownames

rTranscriptome.ave <- data.frame(
  T0 = rowMeans(rTranscriptome[,c(1,9)]),
  T1 = rowMeans(rTranscriptome[,c(2,10)]),
  T6 = rowMeans(rTranscriptome[,c(3,11)]),
  T12 = rowMeans(rTranscriptome[,c(4,12)]),
  T24= rowMeans(rTranscriptome[,c(5,13)]),
  T36 = rowMeans(rTranscriptome[,c(6,14)]),
  T48 = rowMeans(rTranscriptome[,c(7,15)]),
  T72 = rowMeans(rTranscriptome[,c(8,16)]))
  
colnames(rTranscriptome.ave) <- c("0hr", "1hr", "6hr", "12hr", "24hr", "36hr", "48hr", "72hr")
rTranscriptome.ave <- as.matrix(rTranscriptome.ave)
rownames(rTranscriptome.ave) <- rownames(rTranscriptome)

rTranscriptome.ave.unique <- rTranscriptome.ave[unique(rownames(rTranscriptome.ave)),]
```

# Partition into ten ESC expression groups
```{r}
rTranscriptome.ave.unique.percentile <- sort(rTranscriptome.ave.unique[,"0hr"], decreasing=T)

chunk <- function(x,n) split(x, factor(sort(rank(x)%%n)))
groups <- chunk(rTranscriptome.ave.unique.percentile,5)
groups <- sapply(groups, names)

tmp <- lapply(groups, function(x) {
    log(rTranscriptome.ave.unique.percentile[x])
    })

par(mfrow=c(1,1))
boxplot(tmp,
        outline=F,
        ylab = "Gene expression at 0h ESC", 
        xlab = "Expression groups")

exprs10.names <- lapply(tmp, names)
```

# control against PIscore
```{r}
PI <- read.table("../../Data/PI_ES_0h.txt", header=F)
PI <- sapply(split(PI[,3], PI[,2]), mean)
names(PI) <- toupper(names(PI))
PIscore <- sort(PI, decreasing=T)

print("Number of genes")
length(PIscore)

chunk <- function(x,n) split(x, factor(sort(as.integer(rank(x))%%n)))
groups <- chunk(PIscore,10)
groups <- sapply(groups, names)

tmp <- lapply(groups, function(x) {
    log2(PIscore[x])
    })

par(mfrow=c(1,1))
boxplot(tmp,
        outline=F,
        ylab = "Log2 pausing index at 0h ESC", 
        xlab = "Expression groups")
```
# LSD1 signal
```{r}
Lsd1 <- read.delim("../../Data/tssAsBsites_mm9_e500.boxplot_Lsd1", header=F) 

tmp <- sapply(strsplit(as.character(Lsd1$V5), ","), "[[", 2)
genes <- toupper(sapply(strsplit(tmp, ";"), "[[", 1))
Lsd1$V5 <- genes

Lsd1.signal <- Lsd1$V7
names(Lsd1.signal) <- Lsd1$V5
Lsd1.signal <- sapply(split(Lsd1.signal, names(Lsd1.signal)), mean)
```

```{r}
tmp2 <- lapply(groups, function(x) {
    na.omit(Lsd1.signal[x])
    })

boxplot(tmp2,
        outline=F, 
        col=rainbow(10),
        ylab="LSD1 signal (RPM)",
        xlab="Pausing groups")
```
# Density plots partitioned by gene expression group
```{r}
rTranscriptome.ave.unique.percentile <- sort(rTranscriptome.ave.unique[,"0hr"], decreasing=T)

chunk <- function(x,n) split(x, factor(sort(rank(x)%%n)))
groups <- chunk(rTranscriptome.ave.unique.percentile,5)
groups <- sapply(groups, names)

path <- "../../Data/tss_mESC_exprs_Lsd1/"
tagDensity_files <- list.files(path)

par(mfrow=c(1,2))
  
  c <- lapply(tagDensity_files, function(x) read.delim(paste0(path, x), header = F))

  ## curve based fitting
  curve.fit <- function(y, span=0.2, fit=T) {
    if (fit) {
      x <- 1:length(y)
      y.loess <- loess(y ~ x, span=span, data.frame(x=x, y=y))
      y.predict <- predict(y.loess, data.frame(x=1:length(y)))
      return (y.predict)
    } else {
      return (y)
    }
  }
  m <- 1000000
  
  signal <- lapply(c,  function(x) {
        
      curve.fit(x[25:75,'V4'], fit=T)*m
      
  })
  
  a <- min(do.call(rbind, signal), na.rm=T)
  b <- max(do.call(rbind, signal), na.rm=T)
  
  margin <- (b - a) / 20
  a <- a - margin
  b <- b + margin
  
  
names(signal) <- c("p1", "p10", "p2", "p3", "p4", "p5", "p6" ,"p7", "p8", "p9")
signal <- signal[c("p1", "p2", "p3", "p4", "p5", "p6" ,"p7", "p8", "p9","p10")]


par(mfrow=c(1,1))
  plot(signal[[1]], type="l", lwd=2, 
       ylim=c(a, b),
       col=rainbow(10)[[1]],
       cex.main=1, 
       ylab="Lsd1 signal (rpm/bp)", 
       xaxt="n")
  axis(1, at=seq(25, 75, by=25), 
       labels=c("TSS", "2.5kb", "0"))
for (i in 2:10) {
  points(signal[[i]], type="l", lwd=2, 
         ylim=c(a, b),
         col=rainbow(10)[[i]],
         cex.main=1)
 }
```

```{r}
tmp <- lapply(groups, function(x) {
    log(rTranscriptome.ave.unique.percentile[x])
})
  
boxplot(tmp,
        outline=F, 
        col=rainbow(10),
        names=sapply(groups, length),
        ylab="log expression in mESC (0hr)",
        xlab="groups")
```

# Session Info
```{r}
sessioninfo::session_info()
```

